dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ReferenceConfig.java

```diff
@@ -532,17 +532,34 @@ private void createInvokerForRemote() {
 
     private void checkInvokerAvailable() throws IllegalStateException {
         if (shouldCheck() && !invoker.isAvailable()) {
-            invoker.destroy();
-            throw new IllegalStateException("Failed to check the status of the service "
-                + interfaceName
-                + ". No provider available for the service "
-                + (group == null ? "" : group + "/")
-                + interfaceName +
-                (version == null ? "" : ":" + version)
-                + " from the url "
-                + invoker.getUrl()
-                + " to the consumer "
-                + NetUtils.getLocalHost() + " use dubbo version " + Version.getVersion());
+            try {
```

Comment:
It is better to adopt a more general-purpose mechanism to manage `ReferenceConfig`'s lifecycle in avoid of other `Exception` s or rollbacks may occur during initializing phase. Not only `checkAvailable()`

Reply:
Other resources created and shared by `Config`s may also be took into consideration.
- Service Discovery: Instance subscriptions/ URLs 
- Network IO. Netty's EventLoop/ Timer/ Connections.
- Caches